src/replication/replication_messages.cpp

```diff
@@ -86,26 +95,27 @@ RecordsBatchMsg::RecordsBatchMsg(ReplicationMessageMetadata metadata, record_bat
 
 // TxnAppliedMsg
 
-common::json TxnAppliedMsg::ToJson() const {
-  common::json json = BaseReplicationMessage::ToJson();
-  json[key_applied_txn_id] = applied_txn_id_;
-  return json;
+MessageFacade TxnAppliedMsg::ToMessageFacade() const {
+  MessageFacade message = BaseReplicationMessage::ToMessageFacade();
+  message.Put(key_applied_txn_id, applied_txn_id_);
+  return message;
 }
 
-TxnAppliedMsg::TxnAppliedMsg(const common::json &json)
-    : BaseReplicationMessage(json), applied_txn_id_(json[key_applied_txn_id].get<transaction::timestamp_t>()) {}
+TxnAppliedMsg::TxnAppliedMsg(const MessageFacade &message)
+    : BaseReplicationMessage(message), applied_txn_id_(message.Get<transaction::timestamp_t>(key_applied_txn_id)) {}
 
 TxnAppliedMsg::TxnAppliedMsg(ReplicationMessageMetadata metadata, transaction::timestamp_t applied_txn_id)
     : BaseReplicationMessage(ReplicationMessageType::TXN_APPLIED, metadata), applied_txn_id_(applied_txn_id) {}
 
-std::unique_ptr<BaseReplicationMessage> BaseReplicationMessage::ParseFromJson(const common::json &json) {
-  // BaseReplicationMessage switches on the json's key_message_type to figure out what type of message to create.
-  ReplicationMessageType msg_type = ReplicationMessageTypeFromString(json[key_message_type]);
+std::unique_ptr<BaseReplicationMessage> BaseReplicationMessage::ParseFromString(std::string_view str) {
+  MessageFacade message(str);
+  // BaseReplicationMessage switches on the message's key_message_type to figure out what type of message to create.
+  ReplicationMessageType msg_type = ReplicationMessageTypeFromString(message.Get<std::string>(key_message_type));
   switch (msg_type) {
-      // clang-format off
-    case ReplicationMessageType::NOTIFY_OAT:          { return std::make_unique<NotifyOATMsg>(json); }
-    case ReplicationMessageType::RECORDS_BATCH:       { return std::make_unique<RecordsBatchMsg>(json); }
-    case ReplicationMessageType::TXN_APPLIED:         { return std::make_unique<TxnAppliedMsg>(json); }
+    // clang-format off
```

Comment:
If I'm not mistaken, this line here still gets clang-format applied to it. That or CLion ctrl-alt-L format did it. but if CI is happy, I'm happy!

Reply:
Yeah not really sure what happened here, I didn't explicitly move it. I think it was moved due to clang-format.